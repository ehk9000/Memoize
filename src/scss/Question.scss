@import './variables.scss';

.card-container {
    display: $flex;
    flex-flow: column nowrap;
    height: 600px;
    width: 80%;
    border: $border;
    position: fixed;
    justify-content: space-between;
    left: 170px;
    bottom: 150px;
    background-color: $backgroundColor; 
    box-shadow: $boxShadow;
}

input {
    align-self: center;
    width: 50%;
}

.submit {
    margin-bottom: 25px;
}

