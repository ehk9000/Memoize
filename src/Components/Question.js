import React, {Component} from 'react';
import '../scss/Question.scss';



class Question extends Component {
    constructor(props){
        super(props);
        this.state = {
            errorMsg: ''
        }
    }
    

    checkAnswer = (e) => {
        e.preventDefault();
        let input = e.target.closest('.card-container').querySelector('input').value
        if (input.toLowerCase() === this.props.currentQuestion.answer.toLowerCase()) {
            this.setState({
                errorMsg: "Nice!"
            }) 
            setTimeout(() => this.nextQuestion(), 2000);
            
        } else {
            this.setState({
                errorMsg: "Try again!"
            })
        }
    }

    nextQuestion = () => {
         this.setState({errorMsg: ''});
         this.props.removeQuestion();
    }


    render() {  
    if (this.props.currentQuestion === undefined) {
        return null;
    }

        return (
            <article className="card-container">
                <h3>{this.props.currentQuestion.question}</h3>
                <h3>{this.props.currentQuestion.exampleString}</h3>
                <p>{this.state.errorMsg}</p>
                <input type="text" className="user-guess" id="user-guess"/>
                <input type="submit" onClick={this.checkAnswer} className="submit"/>
            </article>
        )
    }
}

export default Question;