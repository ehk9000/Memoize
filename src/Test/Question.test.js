import React from 'react';
import ReactDOM from 'react-dom';
import Question from '../Components/Question.js';
import {shallow} from 'enzyme';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const mockEvent = { preventDefault: jest.fn() }
const mockRemoveQuestion = {
    removeQuestion: jest.fn()}
const testEvent =  {target: { value: 'foo' }} 



const mockQuestion = {
    "question": "How would you split each word into its own string?",
    "exampleString": "let bemo = \"Yes, Finn. It goes in my butt.\"",
    "answer": "bemo.split(\" \")",
    "category": "easy",
    "id": 7
}

describe('Question', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Question 
            currentQuestion={mockQuestion} 
            removeQuestion = {mockRemoveQuestion}/>
        )
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('should have default state', () => {
        expect(wrapper.state()).toEqual({
            errorMsg: ''
        })
    })

    it('should check answer', () => {
        wrapper.find('.submit').simulate('click', mockEvent)
        wrapper.instance().checkAnswer(testEvent)
    })

    it('should fire method to delete question in parent', () => {
        wrapper.instance().nextQuestion();

        expect(mockRemoveQuestion).toHaveBeenCalled()
    })
})

