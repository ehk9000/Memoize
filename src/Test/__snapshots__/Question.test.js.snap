// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match snapshot 1`] = `
<article
  className="card-container"
>
  <h3>
    How would you split each word into its own string?
  </h3>
  <h3>
    let bemo = "Yes, Finn. It goes in my butt."
     
  </h3>
  <p />
  <input
    className="user-guess"
    id="user-guess"
    type="text"
  />
  <input
    className="submit"
    onClick={[Function]}
    type="submit"
  />
</article>
`;
